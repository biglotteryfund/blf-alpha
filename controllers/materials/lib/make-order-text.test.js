/* eslint-env jest */
'use strict';

const makeOrderText = require('./make-order-text');

test('should make order text for email', () => {
    const items = [
        {
            name: 'Stainless steel plaque',
            code: 'BLF-BR088',
            quantity: 1
        },
        {
            name: 'Vinyl banner (pink)',
            code: 'BIG-BANNP',
            quantity: 1
        },
        {
            name: 'Balloons',
            code: 'BIG-EVBLN',
            quantity: 2
        }
    ];

    const details = {
        yourName: 'Ann Example',
        yourEmail: 'ann@example.com',
        yourAddress1: '1 Plough Place',
        yourAddress2: '',
        yourCounty: '',
        yourTown: 'London',
        yourCountry: 'United Kingdom',
        yourPostcode: 'EC4A 1DE',
        yourProjectName: '',
        yourReason: 'projectOpening',
        yourReasonOther: '',
        yourGrantAmount: 'over10k',
        yourGrantAmountOther: ''
    };

    const orderText = makeOrderText(items, details);
    expect(orderText).toContain(
        '- x1 BLF-BR088 (item: Stainless steel plaque)'
    );
    expect(orderText).toContain('- x1 BIG-BANNP (item: Vinyl banner (pink))');
    expect(orderText).toContain('Name: Ann Example');
    expect(orderText).toContain('Email address: ann@example.com');
    expect(orderText).toContain('Postcode: EC4A 1DE');
});
