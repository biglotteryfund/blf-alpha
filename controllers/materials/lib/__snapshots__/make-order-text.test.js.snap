// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should make order text for email 1`] = `
"A new order has been received from The National Lottery Community Fund website.
The order details are below:

- x1 BLF-BR088 (item: Stainless steel plaque)
- x1 BIG-BANNP (item: Vinyl banner (pink))
- x2 BIG-EVBLN (item: Balloons)

The customer's personal details are below:

Name: Ann Example
Email address: ann@example.com
Address line 1: 1 Plough Place
Town/city: London
Country: United Kingdom
Postcode: EC4A 1DE
Grant amount: over10k
Order reason: projectOpening

This email has been automatically generated from The National Lottery Community Fund website.
If you have feedback, please contact digital.monitoring@tnlcommunityfund.org.uk."
`;
